tests/test_run.py

```diff
@@ -54,6 +58,49 @@ def test_MDRunner():
     assert rc == 0, "mdrun failed to run through MDrunner"
 
 
+def test_MDRunner_keyboard_interrupt(monkeypatch):
+    """Test that a keyboard interrupt is handled correctly."""
+
+    # Create a mock for subprocess.Popen
+    class MockPopen:
+        def __init__(self, *args, **kwargs):
+            pass
+
+        def wait(self):
+            time.sleep(2)  # Simulate a long-running process
+
+        def terminate(self):
+            pass
+
+    # Use monkeypatch to replace subprocess.Popen with MockPopen
+    monkeypatch.setattr("gromacs.run.subprocess.Popen", MockPopen)
+
+    # Initialize MDrunner
+    mdrunner = gromacs.run.MDrunner()
+
+    # Function to send SIGINT after a delay
+    def send_interrupt():
+        time.sleep(1)  # Short delay before sending SIGINT
+        os.kill(os.getpid(), signal.SIGINT)
+
+    # Start a thread to send the SIGINT
+    interrupt_thread = threading.Thread(target=send_interrupt)
```

Comment:
That is nifty/hacky. Did the robot suggest it?

Reply:
The robot did indeed suggest it but it took a lot of prompting :)